src/masternodes/consensus/tokens.cpp

```diff
@@ -107,14 +112,136 @@ Res CTokensConsensus::operator()(const CMintTokensMessage& obj) const {
         if (!mintable)
             return std::move(mintable);
 
-        auto minted = mnview.AddMintedTokens(tokenId, kv.second);
+        if (static_cast<int>(height) >= consensus.GreatWorldHeight && token->IsDAT() && !HasFoundationAuth())
+        {
+            mintable.ok = false;
+
+            auto attributes = mnview.GetAttributes();
+
+            CDataStructureV0 membersKey{AttributeTypes::Token, tokenId.v, TokenKeys::ConsortiumMembers};
+            auto members = attributes->GetValue(membersKey, CConsortiumMembers{});
+
+            CDataStructureV0 membersMintedKey{AttributeTypes::Live, ParamIDs::Economy, EconomyKeys::ConsortiumMembersMinted};
+            auto membersBalances = attributes->GetValue(membersMintedKey, CConsortiumMembersMinted{});
+
+            for (auto const& tmp : members)
+            {
+                auto key = tmp.first;
+                auto member = tmp.second;
+
+                if (HasAuth(member.ownerAddress))
+                {
+                    if (membersBalances[key].balances[tokenId] + amount > member.mintLimit)
+                        return Res::Err("You will exceed your maximum mint limit for %s token by minting this amount!", tokenImpl.symbol);
```

Comment:
It's better to be done after `Add`, so check result of `Add` then we can introduce `Get` in `CBalances` so we can check the result against mint limit safety.